### DISCUSSION TITLE
SRE not working

### DISCUSSION THREAD

**Discussion by cat6man** [OpenSageTV/Community#123] [Other Plugins]
hi folks,
it's baseball season and i don't see SRE working.  i've forgotten what setting or url i'm supposed to point to these days.
any help would be appreciated to show me the way to resurrect the BEST PLUGIN EVER!
thanks

**Comment by wayner9** [comment]
It is working for me, at least for an NBA game that is going on right now.  But not for the NCAA basketball but that might be due to a different show title - 2024 NCAA Men's Basketball Tournament rather than COLLEGE BASKETBALL.

What do you see on your LivePVR Data web page?  Go to your sage IP/port number and you should see a link to Live PVR Data server.

I see this:
	COLLEGE BASKETBALL
	==================
	NC State vs. Duke                                 	12:02 - 2nd        
	Tennessee vs. Purdue                              	Final              

	COLLEGE FOOTBALL
	================
	No games for this date.

	MLB BASEBALL
	============
	Guardians vs. Athletics                           	Bot 9th            
	Red Sox vs. Mariners                              	Bot 8th            
	Rockies vs. Diamondbacks                          	Top 9th            
	Giants vs. Padres                                 	Bot 7th            
	Angels vs. Orioles                                	Final              
	Braves vs. Phillies                               	Final              
	Brewers vs. Mets                                  	Final              
	Pirates vs. Marlins                               	Final/10           
	Blue Jays vs. Rays                                	Final              
	Nationals vs. Reds                                	Final              
	Tigers vs. White Sox                              	Final              
	Twins vs. Royals                                  	Final              
	Yankees vs. Astros                                	Final              
	Cubs vs. Rangers                                  	Final              
	Cardinals vs. Dodgers                             	Mar 31 19:00 ET    

	FORMULA ONE RACING
	==================
	No games for this date.

	NASCAR RACING
	=============
	No games for this date.

	NBA BASKETBALL
	==============
	Lakers vs. Nets                                   	3:16 - 1st         
	Clippers vs. Hornets                              	End of 1st         
	76ers vs. Raptors                                 	1:28 - 1st         
	Heat vs. Wizards                                  	31.8 - 1st         
	Cavaliers vs. Nuggets                             	Final              
	Thunder vs. Knicks                                	Mar 31 19:00 ET    
	Mavericks vs. Rockets                             	Mar 31 19:00 ET    
	Bulls vs. Timberwolves                            	Mar 31 19:00 ET    
	Warriors vs. Spurs                                	Mar 31 19:00 ET    
	Jazz vs. Kings                                    	Mar 31 21:00 ET    

	WNBA BASKETBALL
	===============
	No games for this date.

	NFL FOOTBALL
	============
	No games for this date.

	NHL HOCKEY
	==========
	Ducks vs. Canucks                                 	Final              

	COLLEGE SOFTBALL
	================
	Arkansas vs. Georgia                              	Final              
	South Carolina vs. Ole Miss                       	Final              
	Northwestern vs. Ohio State                       	Final              
	Maryland vs. Penn State                           	Final/12           
	Michigan vs. Indiana                              	Final/5            
	Louisville vs. Pitt                               	Final              
	Tennessee vs. Auburn                              	Final              
	Nebraska vs. Wisconsin                            	Final/9            

	COLLEGE BASEBALL
	================
	Maryland vs. Michigan                             	Final              
	Mississippi St vs. Florida                        	Final              
	North Carolina vs. Wake Forest                    	Final              
	Nebraska vs. Northwestern                         	Final/10           
	Purdue vs. Ohio State                             	Final              
	Oklahoma St vs. West Virginia                     	Final              
	Minnesota vs. Iowa                                	Final              
	Rutgers vs. Michigan St                           	Final              
	Georgia vs. Tennessee                             	Final              
	Kentucky vs. Ole Miss                             	Final/7            

	WOMEN'S COLLEGE BASKETBALL
	==========================
	Oregon St vs. South Carolina                      	Final

**Reply by trk2** [reply]
Everything is working on my SRE, including MLB and college basketball:

	COLLEGE BASKETBALL
	==================
	NC State vs. Duke                                 	12:02 - 2nd        
	Tennessee vs. Purdue                              	Final              

	COLLEGE FOOTBALL
	================
	No games for this date.

	MLB BASEBALL
	============
	Cardinals vs. Dodgers                             	Bot 4th            
	Angels vs. Orioles                                	Final              
	Braves vs. Phillies                               	Final              
	Brewers vs. Mets                                  	Final              
	Pirates vs. Marlins                               	Final/10           
	Blue Jays vs. Rays                                	Final              
	Nationals vs. Reds                                	Final              
	Tigers vs. White Sox                              	Final              
	Twins vs. Royals                                  	Final              
	Yankees vs. Astros                                	Final              
	Cubs vs. Rangers                                  	Final              
	Guardians vs. Athletics                           	Final              
	Red Sox vs. Mariners                              	Final              
	Rockies vs. Diamondbacks                          	Final              
	Giants vs. Padres                                 	Final

**Comment by emveepee** [comment]
The NextPVR implementation which uses the same server is still working checking a Blue Jays game from earlier today

LivePvrData.Data.Net.Responses.StatusResponse [Title=MLB Baseball, Status=Final, IsValid=True, IsActive=False, IsComplete=True, IsError=False, ]

Perhaps you aren't using the updated server from last year when MLB changed?

**Reply by cat6man** [reply]
@emveepee 

that sounds like the problem, as this started last year :(
since my sageTV chops have deteriorated over the years, where is the update you refer to and how do i install it?
i'm running the latest ubuntu64 bit

@wayner9 
i remember slugger well.  his contributions were wonderful.

thanks for the help people.  i really appreciate it.

**Reply by emveepee** [reply]
@cat6man the update is here with a war file https://github.com/livepvrdata-oss/livepvrdata-svc/releases/tag/2.1.0

**Reply by wayner9** [reply]
Can you not just update via the SageTV Plugin manager - Setup/SageTV Plugins/Installed Plugins and then click on General and go to Live PVR Data Server.

![image](https://github.com/OpenSageTV/Community/assets/10366145/d2d14b3e-d599-4e1f-8dba-d5e655545f0d)

**Comment by emveepee** [comment]
> It is working for me, at least for an NBA game that is going on right now. But not for the NCAA basketball but that might be due to a different show title - 2024 NCAA Men's Basketball Tournament rather than COLLEGE BASKETBALL.

You are very close.  The reason it isn't working  is that Gracenote have added "Men's" and "Women's" into the title this year and so the regex here https://github.com/livepvrdata-oss/livepvrdata-svc/blob/04508a247a4b659d1090d624d171928112401eaa/src/main/groovy/com/github/livepvrdata/MonitorFactory.groovy#L50 fails .  There was never a match for the Women's Tournament before.  If you guys build your own change MonitorFactory.groovy as follows

```
			case ~/College Basketball|\d{4} NCAA Men's Basketball Tournament/:
...
			case ~/Women's College Basketball|\d{4} NCAA Women's Basketball Tournament/:
```

**Reply by wayner9** [reply]
Slugger from Kingston was the guy that wrote SRE, as well as SJQ, SageAlert, and some other plugins as well.  When he left SageTV it was a big loss.  This was such an intelligent plugin.

**Reply by emveepee** [reply]
For sure Slugger was helpful with my port for NextPVR too but the issue with SageTV is much bigger.  It isn't just Slugger almost all developers left.  skiingwiz did continue to support SRE  and I could submit a PR, and maybe it will get fixed, but you guys need active support.

**Reply by wayner9** [reply]
The stuff that Slugger did with SJQ and Groovy scripting was very powerful.  I played around with the customized Groovy web pages and created a page to monitor all of my extenders, giving the status of all of them, including current playback file and time, and the ability to reboot the extenders with telnet commands.

But I think I was one of only a handful of people that used it extensively.  It seems like a lot of the SageTV developers were Canadian - Slugger, Jusjoken, Stuckless, etc.

**Comment by jbuszkie** [comment]
Yes..  I'm digging up an old thread VS starting a new one!

My SRE stopped working (I don't know when)
Is it still working for you guys?
The livePVR server stops at just printing "College Basketball"
I see this in the jettystarter.log
2026-09-09 18:41:39,080 [qtp1359466180-123] WARN  org.eclipse.jetty.server.HttpChannel - handleException /livepvrdata/events.groovy java.io.IOException: Server returned HTTP response code: 403 for URL: http://site.api.espn.com/apis/site/v2/sports/basketball/mens-college-basketball/scoreboard/conferences?groups=50

But if I paste the URL in a browser it works.

Any thoughts?

Jim

**Reply by mickgeller** [reply]
I'm glad you are seeing the same exact thing I've been seeing since sometime in mid-august. I convinced myself that it was only me since no one else had reported it. Go figure.

I record baseball games. I've changed my favorites to add an hour to the end of a recording and that has worked for me well so far. I dug into it two or three weeks ago. I see what you're seeing with the 403 message but could not figure out next steps to resolve it.

I thought it might have been a random IP address blockage. I am on unraid, but I resurrected a Windows instance and used a VPN to test. I got the same results with that setup, so I guess it's not an IP address blockage.

The final thing I noted, on the current problem, is that the SRE logs do not indicate any issues at all. The only issue seen is in the jetty starter log.

With football season starting, my guess is that a lot more people will notice it. Good luck. I don't record any football games, usually. YMMV

**Reply by wayner9** [reply]
I see the same:
	COLLEGE BASKETBALL
	==================

**Comment by emveepee** [comment]
All the ESPN API monitors broke a few weeks ago, they need specific user agent and the default Java one doesn't work.  MLB still uses mlb.com.

**Reply by wayner9** [reply]
Maybe we could use AI to help rewrite the plugin?

**Reply by jbuszkie** [reply]
Actually MLB doesn't use mlb.com anymore.  According to claude.  
From Claude:
It was commit bacfee2 — "Move to other MLB monitor. Current one is broken." by skiingwiz on March 22, 2023.

That commit switched MLB from the dedicated EventMonitorMLB (which uses MLB's own API at mlb.com) to the ESPN-based MLB monitor. The EventMonitorMLB.groovy file is still in the source tree but isn't wired up in MonitorFactory anymore. So MLB data now comes through ESPN like everything else — which means the User-Agent fix we just applied covers it too.

**Comment by jbuszkie** [comment]
yeah it's weird that the url posted in a browser works..

**Comment by jbuszkie** [comment]
OK Claude fixed it for me!  It's working again. It had to recompile 7 classes.
[SRE_fix.zip](https://github.com/user-attachments/files/32078783/SRE_fix.zip)

Now the real fix is to redeploy in the plugin.  But I don't know enough to know how to do that..  I'm sure I could eventually figure it out with Claude or Cursor...

For me I had to copy the files (overwriting) into 
jetty\work\jetty-0_0_0_0-8081-livepvrdata_war-_livepvrdata-any-\webapp\WEB-INF\classes\com\github\livepvrdata\monitors\espn\json

I made a copy of the json dir in case I have to go back.

I don't know how or when or if they ever get overwritten from the WAR file..  So that may need to be updated too!
<img width="635" height="853" alt="image" src="https://github.com/user-attachments/assets/f9d1f32f-43cb-42bf-a9b4-162efaa180df" />

**Comment by jbuszkie** [comment]
Summary from Claude:

**Fix for ESPN 403 errors in livepvrdata / SRE**

The ESPN scoreboard API started returning 403 Forbidden responses, which broke game status lookups in livepvrdata (used by the SRE plugin).

Root cause: Monitor.groovy and EspnMonitorUtils.groovy werefetching ESPN's API using bare new URL(url).text calls with no HTTP headers. ESPN now rejects requests without a User-Agent.

Fix: Added a fetchUrl() helper method to Monitor.groovy that uses HttpURLConnection with browser-like headers (User-Agent, Accept, Accept-Language) and 15-second timeouts. Both files now call this instead of new URL(url).text.

To apply: Replace these 7 class files in your Jetty work directory under work/jetty-0_0_0_0-8081-livepvrdata_war-_livepvrdata-any-/webapp/WEB-INF/classes/com/github/livepvrdata/monitors/espn/json/:

Monitor.class
Monitor$_execute_closure3.class
Monitor$_getEventsForDate_closure2.class
Monitor$_parseFeed_closure1.class
Monitor$_parseFeed_closure1_closure4.class
EspnMonitorUtils.class
EspnMonitorUtils$_fetchGroupIds_closure1.class

Then restart SageTV.

The patched source files and compiled classes are available if anyone needs them.

**Reply by wayner9** [reply]
I have a slightly different directory on Unraid which may be due to a different version of Jetty or Java:

appdata\sagetv\server\jetty\work\jetty-0_0_0_0-80-livepvrdata_war-_livepvrdata-any-\webapp\WEB-INF\classes\com\github\livepvrdata\monitors\espn\json

**Reply by jbuszkie** [reply]
Probably my port is 8081....  And your is 80?

**Reply by wayner9** [reply]
Yes, that explains it.

**Reply by mickgeller** [reply]
The Jetty Web Server has an option on its config to Restart the Web Server. Using that worked for me without restarting Sage. I had some recordings going and did not want to wait for them to finish. 

Thanks to one and all for the fix.

**Reply by jbuszkie** [reply]
Good to know!  I wasn't in front of any of my boxes to check!  That would save some time while testing!

**Reply by wayner9** [reply]
FYI - to restart the Jetty Server you go into the Plugin under Settings-Plugins-Installed Plugins. Click on Jetty Web Server and then choose Configure Plugin and it should be at the top.

**Reply by KeithAbb0tt** [reply]
It's too bad there isn't an option for this on the webgui itself...

**Reply by wayner9** [reply]
I am pretty sure that it can be added. In one of the last updates to the Jetty Web Server Slugger added in the ability to add items to the SageTV Menu.  I used that to add some stuff a long time ago, but I can't remember it right now. I wonder if I could vibe code that change.

I created a SageTV Client manager to allow you to see more info on clients, like extenders and PS, that used this feature
<img width="889" height="537" alt="image" src="https://github.com/user-attachments/assets/c71c75c9-b2ab-4059-91f1-be9f76c49739" />

**Reply by KeithAbb0tt** [reply]
Thanks, @wayner9!  I checked in the webgui under SageTV / Plugin Manager / Configure Plugin / Jetty Web Server / Configure and, lo and behold, the option to "Restart Web Server" is listed.  Still learning new things after this many years.

btw, sweet looking client manager.

**Reply by wayner9** [reply]
I noticed one issue when restarting the server without restarting the SageTV service/docker - some apps may no longer work. The SageX API Jetty page was not accessible until I did a restart.
The client manager is functional but it looks kind of like a 1996 web page, but I think that is due to the style of the SageUI.  I will see if I can vibe code a better looking UI.

**Comment by jbuszkie** [comment]
<img width="1121" height="941" alt="image" src="https://github.com/user-attachments/assets/bac29bb2-2d39-4d71-8873-92b203a3c958" />
I've been playing with rebuilding the livePVR Data server.  Cursor built a new webpage! lol
It has a bunch more information on it.  I think I'll get rid of most and just keep the scoreboard.
I have a war file if anyone is interested.

**Reply by wayner9** [reply]
Looks awesome.  Great job!

**Reply by wayner9** [reply]
Just copied over the new files, restarted the Jetty server and it is now working.

AI vibe coding is helping to keep SageTV running!

**Comment by cat6man** [comment]
That's great but not sure how often I'll use it.  Verizon FiOS killed cable
card and my Prime no longer works.
So, I put in a VoIP phone and cancelled both phone and cable tv, saving
almost $200 a month (minus $80 a my nth for youtube tv).  I still have
sagetv running (client on shield pro) and would/will(?) use it for ota next
Gen if the broadcasters weren't encrypting it.
Next step was to sign up for streaming MLB but they had a deal with ESPN
that required me to create a Disney login to sign up.  Screw em, I'm
listening to MLB game day audio and enjoying that more than sitting in
front of a tv set.
So I'm not very active here anymore but still hanging on by my fingernails
to sagetv.

Just wondering, can sagetv record youtube?

The other thing I miss is native resolution pass through, which seems not
to be supported on the shield pro.  I used to be se that all the time with
a dvdo scaler since the early tv upscalers were pretty ugly.  Nowadays tv
pixel scaling is far ahead of the streamers but without native passthrough
we can't use it.

Peace

**Reply by jbuszkie** [reply]
My cable card through xfinity still works..  So far.  But when that's done, I'm done.  We only use sage for the news and sporting events.  Everything else is streaming! I still have a million shows on Sage that I'll probably never watch!  Maybe the wife still has shows on regular TV she watches..  But I dont!

**Reply by wayner9** [reply]
For the most part I just watch sports. I really want to get it working for IPTV as that is what I use as my TV source now. I really like being able to timeshift sports, even to just be able to skip through commercials, halftime, etc. SageTV was SO good for that with customizable skip times, etc.

**Reply by jbuszkie** [reply]
Yes!  I still love the comskip and FF 10 or 30 sec as I have it set!

**Reply by wayner9** [reply]
I use 10 seconds and 120 or 150 seconds to get through commercial breaks. I use the dynamic skip time plugin to optimize for the sport.
